### PR TITLE
add auto upgrade mapping logic

### LINKED ISSUE #677: Add schema_version to .opensearch-notifications-config index mappings and automatically upgrade logic
**Is your feature request related to a problem?**
Currently we only have one version of mappings for .opensearch-notifications-config index. However, as we plan to support more notification channels, we will have new mappings. Current code doesn't check the version of index mappings, neither update the mappings. This will cause problems for clusters upgraded from older notifications versions.

**What solution would you like?**
Like ism plugin, add schema_version to .opensearch-notifications-config index mappings and automatically upgrade logic. The upgrade logic is triggerred by CRUD request of .opensearch-notifications-config.

**What alternatives have you considered?**
N/A

**Do you have any additional context?**
N/A

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
add field schema_version to notification index mapping, and autoupgrade logic

### Issues Resolved
#677 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/notifications/src/test/kotlin/org/opensearch/integtest/config/CreateNotificationConfigIT.kt [hunk 1]
```diff
@@ -217,4 +220,59 @@ class CreateNotificationConfigIT : PluginRestTestCase() {
...
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        """.trimIndent()
```
Comment (Hailong-am) [inline notifications/notifications/src/test/kotlin/org/opensearch/integtest/config/CreateNotificationConfigIT.kt:261]: can we have a test case to cover `_meta ` is empty which is same as current index mapping?

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt [hunk 2]
```diff
@@ -92,15 +107,28 @@ internal object NotificationConfigIndex : ConfigOperations {
...
+            if (metaData.containsKey(SCHEMA_VERSION)) {
+                try {
+                    schemaVersion = metaData[SCHEMA_VERSION] as Int
+                } catch (e: Exception) {
+                    throw IllegalArgumentException("schema_version can only be Integer")
+                }
+            }
+        }
+        return schemaVersion
+    }
```
Comment (Hailong-am) [inline notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:123]: should the value of `DEFAULT_SCHEMA_VERSION` be different with the [value](https://github.com/opensearch-project/notifications/pull/699/files#diff-19f7d0c225db1c9f27a0003ac8d1c4f50f9b0e39d8e7a6504b19fd90cd36a4f6R11) in yml file?

Comment (zhichao-aws) [reply notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:123]: The DEFAULT_SCHEMA_VERSION is for case that there is no schema_version in index_mappings, i.e. current mappings version. In this case we don't need to update the mappings. These codes will be used after Microsoft Teams PR, which has schema_version=2 .

File: notifications/notifications/src/test/kotlin/org/opensearch/integtest/config/CreateNotificationConfigIT.kt (hunk 1 above)
Comment (zhichao-aws) [reply notifications/notifications/src/test/kotlin/org/opensearch/integtest/config/CreateNotificationConfigIT.kt:261]: sure

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt (hunk 2 above)
Comment (Hailong-am) [reply notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/NotificationConfigIndex.kt:123]: make sense.

**Review by Hailong-am — APPROVED** [review]

**Review by gaobinlong — APPROVED** [review]

**Pull request merged**